src/plugins/intel_gna/samples/transformer/src/app/menu.cpp

```diff
@@ -0,0 +1,173 @@
+
+std::string output_model_message = "Path to file to export transformend model with .xml extension.";
+DEFINE_string(output_model, "", output_model_message.c_str());
+
+static const std::string list_app_transformations_message = "List application defined transformations.";
+DEFINE_bool(list_app_transformations, false, list_app_transformations_message.c_str());
+
+const std::string Menu::kWeightsFileExtension = ".bin";
+constexpr char Menu::kFileNameDelimiter;
+
+const std::string example_command_config =
+    "-input_model model.xml -output_model transformed_model.xml -config configuration.json";
+const std::string example_command_transformation =
+    "-input_model model.xml -output_model transformed_model.xml -transformations transformation_name";
+
+Menu::Menu(int argc, char* argv[]) {
+    init(argc, argv);
+}
+
+static void print_usage() {
+    log_info() << std::endl;
+    log_info() << "gna_transformer_app [OPTION]" << std::endl;
+    log_info() << "Options:" << std::endl;
+    log_info() << std::endl;
+    log_info() << "    -h                            " << help_message << std::endl;
+    log_info() << "    -input_model \"<path>\"       " << input_model_message << std::endl;
+    log_info() << "    -output_model \"<path>\"      " << output_model_message << std::endl;
+    log_info() << "    -config \"<path>\"            " << config_message << std::endl;
+    log_info() << "    -transformation \"<string>\"  " << transformation_message << std::endl;
+    log_info() << "    -list_app_transformations     " << list_app_transformations_message << std::endl;
+    log_info() << std::endl;
+    log_info() << "    commands examples:" << transformation_message << std::endl;
+    log_info() << std::endl;
+
+    log_info() << example_command_config << std::endl;
+    log_info() << example_command_transformation << std::endl;
+}
+
+void Menu::execute_action() {
+    if (m_action) {
+        m_action();
+        log_info() << "Transformation successed" << std::endl;
+    }
+}
+
+void Menu::init(int argc, char* argv[]) {
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        print_usage();
+        return;
+    }
+
+    if (FLAGS_list_app_transformations) {
+        print_app_defined_transformations();
+        return;
+    }
+
+    if (FLAGS_input_model.empty()) {
+        throw WrongArgumentsExeception("Lack of input_model argument. Please check -h option");
+    }
+
+    if (FLAGS_output_model.empty()) {
+        throw WrongArgumentsExeception("Lack of output_model argument. Please check -h option");
+    }
+
+    if (!FLAGS_config.empty()) {
+        m_action = prepare_gna_specific_transformations(FLAGS_config, FLAGS_input_model, FLAGS_output_model);
+        return;
+    }
+
+    if (!FLAGS_transformation.empty()) {
+        m_action = prepare_app_specific_transformation(FLAGS_transformation, FLAGS_input_model, FLAGS_output_model);
+        return;
+    }
+
```

Comment:
add checking that two options are not used at the same time FLAGS_config  FLAGS_transformation